packages/pixeloven-webpack/bundler/src/Bundler.ts

```diff
@@ -1,168 +1,77 @@
+        const fileReporter = await FileReporter({
+            errorOnWarning,
+            outputPath,
+            warnAfterBundleGzipSize: WARN_AFTER_BUNDLE_GZIP_SIZE,
+            warnAfterChunkGzipSize: WARN_AFTER_CHUNK_GZIP_SIZE,
+        });
+        if (options.clean) {
+            logger.info("cleaning up previous builds...");
+            createOrEmptyDir(outputPath);
+        }
+        return new Promise<number>((resolve, reject) => {
+            webpackCompiler.run((err: Error, stats: Stats) =>
+                err ? reject(err) : resolve(fileReporter.fromStats(stats)),
+            );
+        });
     }
 
-    /**
-     * Build client code path
-     */
-    public async client() {
-        createOrEmptyDir(this.config.outputPath);
-        logger.info("Creating an optimized production build...");
-        if (this.compiler.hasClientCodePath) {
-            logger.info("Client code path found...");
-            if (this.compiler.client) {
-                logger.info("Compiling client...");
-                const previousFileSizes: string[] = await measureFileSizesBeforeBuild(
-                    this.clientPath,
-                );
-                return this.handler(
-                    this.clientPath,
-                    previousFileSizes,
-                    this.compiler.client,
-                );
+    async function client() {
+        if (compiler.client) {
+            if (compiler.hasServerCodePath) {
+                logger.info(`client code path has been found...`);
+                return runner(compiler.client, `${options.outputPath}/public`);
             }
-            throw new Error(`Client compiler failed to initialize.`);
+            logger.error(`compiler is set but code path could not be found`);
+            return 1;
         }
-        throw new Error(`Client code path ${this.clientPath} does not exist.`);
+        return 0;
     }
 
-    /**
-     * Build server code path
-     */
-    public async server() {
-        if (this.compiler.hasServerCodePath) {
-            logger.info("Server code path found...");
-            if (this.compiler.server) {
-                logger.info("Compiling server...");
-                const previousFileSizes: string[] = await measureFileSizesBeforeBuild(
-                    this.serverPath,
-                );
-                return this.handler(
-                    this.serverPath,
-                    previousFileSizes,
-                    this.compiler.server,
-                );
+    async function server() {
+        if (compiler.server) {
+            if (compiler.hasServerCodePath) {
+                logger.info(`server code path has been found...`);
+                return runner(compiler.server, options.outputPath);
             }
-            throw new Error(`Server compiler failed to initialize.`);
+            logger.error(`compiler is set but code path could not be found`);
+            return 1;
```

Comment:
What if client returned 1 and server returned 2 on error, then we'd be able to tell which failed if there's only one.

Reply:
Updated. So now when this returns to the CLI 1=client failure, 2=server failure and 3 would be both failed. We should probably update the status code handling in the CLI. I might try and do that now actually